src/Compression/CompressionCodecDoubleDelta.cpp

```diff
@@ -163,9 +163,8 @@ inline Int64 getMaxValueForByteSize(Int8 byte_size)
         case sizeof(UInt64):
             return std::numeric_limits<Int64>::max();
         default:
-            assert(false && "only 1, 2, 4 and 8 data sizes are supported");
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "only 1, 2, 4 and 8 data sizes are supported");
```

Comment:
```suggestion
            throw Exception(ErrorCodes::LOGICAL_ERROR, "only 1, 2, 4 and 8 data sizes are supported");
```

If this function is called anything else than the supported values, then a logical error happened somewhere. `BAD_ARGUMENTS` error is not something we consider harmful, but here we want to have something noticeable (e.g. `assert(false)` as before). Or maybe use `chassert` which also triggered in non debug builds.